webhooks/clusterclass.go

```diff
@@ -123,8 +140,126 @@ func (webhook *ClusterClass) validate(old, new *clusterv1.ClusterClass) error {
 	// Ensure spec changes are compatible.
 	allErrs = append(allErrs, check.ClusterClassesAreCompatible(old, new)...)
 
+	// Ensure no MachineDeploymentClass currently in use has been removed from the ClusterClass.
+	allErrs = append(allErrs, webhook.validateLivingMachineDeploymentClassesNotRemoved(old, new)...)
+
 	if len(allErrs) > 0 {
 		return apierrors.NewInvalid(clusterv1.GroupVersion.WithKind("ClusterClass").GroupKind(), new.Name, allErrs)
 	}
 	return nil
 }
+
+func (webhook *ClusterClass) validateLivingMachineDeploymentClassesNotRemoved(old, new *clusterv1.ClusterClass) field.ErrorList {
+	var allErrs field.ErrorList
+	if old == nil {
+		return allErrs
+	}
+	missingClasses := webhook.removedMachineClasses(old, new)
+
+	// If no classes have been removed return early as no further checks are needed.
+	if len(missingClasses) == 0 {
+		return allErrs
+	}
+
+	// TODO: The first part of this method will need to be broken out to properly account for changes when also validating variables.
```

Comment:
Which part exactly?

Reply:
Everything above this line really - getting the clusters if there is some change in the classes. We'll also need to get the clusters if there's been some change in the variables, so it'd be good to have one call gated by an OR on these two conditions. 

Not necessary for this PR, but wanted to call it out for discussion as we'll want to limit the API calls where possible.

Reply:
All lines above this one - I'm removing this todo from the PR as it's more of a reminder for myself.